boards/shields/lcd_par_s035/boards/frdm_mcxn947_mcxn947_cpu0.overlay

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/{
+	lvgl_pointer {
+		compatible = "zephyr,lvgl-pointer-input";
+		input = <&gt911_lcd_par_s035>;
+		swap-xy;
+		invert-y;
+	};
+};
+
+&flexcomm2_lpi2c2 {
+	status = "okay";
+	gt911_lcd_par_s035: gt911-lcd_par_s035@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+		irq-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
+		alt-addr = <0x14>;
+	};
+};
```

Comment:
Ideally there should be no board specific overlay? If the shield as a touch controller, it should be in the shield overlay. The irq-gpios should also be made generic and rely on a GPIO nexus so that "gpio4 6" is not hard-coded but instead references a physical pin on whatever connector is used to interface the shield with the board.
Thanks!

Reply:
That was our first take and we added the GPIO directly in the dtsi of the board and the refenced in the shield overlay.
The issue is that on MCXn974 the reset pin of the LCD and the touch controller are shared so we would end up ( i fear) having to write this specific overlay anyway.
To be more clear: a generic implementation would not have
`alt-addr = <0x14>;`
but another GPIO for reset
That is why we ended up with this solution and given it is board specific we specified the GPIO directly here.
We can put the GPIO into the mcx dts(i) and reference it here, but we would still need this specif board overlay.
Do you see my point?

Reply:
It's one thing to have a board specific overlay for the thing that are really unique to the board, but it's another to have the entire definition associated with the touch controller be 100% board specific. This kind of defeats the purpose of having shields in the first place :)

As the physical interface between the shield and the host board is a 28-pin FlexIO header, this header should be defined as a GPIO nexus, all the boards having such a connector  (ex. MCXN947) should indicate what actual GPIOs are routed to each of the 28-pins, and the (generic) shield overlay should then once and for all indicate that the reset pin is pin #7 of the GPIO Nexus, as this will always be true.

Similarly, and sorry that I didn't catch it yesterday, the *generic* shield overlay should not make any mention of `flexcomm2_lpi2c2`, which is board specific, but instead attach the touch controller to something like `flexio_i2c`, which each board would define similarly to how boards with arduino headers inidcate which I2C bug is exposed as `arduino_i2c`. 

BTW: it would be great if someone from NXP could also send a PR adding Arduino and MikroE headers definition for the FRDM MCXN947, i.e. something similar to https://github.com/zephyrproject-rtos/zephyr/blob/763dba59d576caf298c0ec5b826a1db58be976b8/boards/nxp/lpcxpresso55s06/lpcxpresso55s06_common.dtsi#L68-L117 :)

Reply:
Thanks a lot fot the pointers!
I will write it in that way and push it when ready

Reply:
@kartben I hope I got it more as it should ( I will split the commit into more reasonable chunks as soon as the content is right) :)